src/transaction-manager.ts

```diff
@@ -0,0 +1,230 @@
+   */
+  async enqueueTransaction(tx: PopulatedTransaction) {
+    // wait until queue is not full by periodically check the queue
+    while (this.waitingTransaction.isFull()) {
+      await new Promise((resolve) => setTimeout(resolve, 1000))
+    }
+    // enqueue the tx to the waiting list
+    this.waitingTransaction.enqueue(tx)
+    // start the transaction manager if not running
+    if (!this.running) this.start()
+  }
+
+  /**
+   * Send the transaction in the waiting list, append it into pendingList
+   */
+  private async sendTransactions() {
+    while (this.pendingTransaction.size < this.maxPendingTxs) {
+      if (this.waitingTransaction.isEmpty()) break
+      const txData = this.waitingTransaction.dequeue()
+      const tx = await this.publishTx({
+        ...txData,
+        nonce: this.nonce,
+      })
+      this.nonce++
+      this.pendingTransaction.add(tx.hash)
+    }
+  }
+
+  private async publishTx(tx: TransactionRequest, bumpFeesImmediately: boolean = false): Promise<TransactionResponse> {
+    let res = undefined
+    let counter = 0;
+    while(true) {
+      if (counter >= MAX_RESEND_LIMIT) {
+        throw new Error(`failed to publish tx: max resend limit(${MAX_RESEND_LIMIT}) reached`)
+      }
+
+      try {
+        if (bumpFeesImmediately) {
+          tx = this.increaseGasPrice(tx)
+        }
+        res = await this.wallet.sendTransaction(tx)
+        break
+      } catch (e) {
+        if (e.message.includes("transaction replacement is underpriced") || e.message.includes("transaction is underprice")) {
+          // this case happen when the tx is already sent before
+          // increase the gas price at next loop
+        } else {
+          // othewise, throw error
+          throw new Error(`failed to publish tx: ${e.message}`)
+        }
+      }
+      bumpFeesImmediately = true // bump fees next loop
+      counter++
+    }
+    return res;
+  }
+
+  increaseGasPrice(tx: TransactionRequest): TransactionRequest {
+    if (tx.gasPrice) {
+      tx.gasPrice = (tx.gasPrice as BigNumber).mul(1.1);
+    }
+    if (tx.maxPriorityFeePerGas) {
+      tx.maxPriorityFeePerGas = (tx.maxPriorityFeePerGas as BigNumber).mul(1.1);
+    }
+    if (tx.maxFeePerGas) {
+      tx.maxFeePerGas = (tx.maxFeePerGas as BigNumber).mul(1.1);
+    }
+    return tx
+  }
+
+  /**
+   * Remove the pending transaction that has been confirmed
+   */
+  async removePendingTxs() {
+    const txs = Array.from(this.pendingTransaction)
+    const currentBlock = await this.wallet.provider.getBlockNumber()
+    const receipts = await Promise.all(
+      txs.map((tx) => this.wallet.provider.getTransactionReceipt(tx))
+    )
+    receipts
+      .filter((receipt) => receipt.blockNumber + 1 <= currentBlock)
+      .forEach((tx) => {
+        // notify the subscriber
+        this.notifySubscribers(tx)
+        // remove the tx from pending list
+        this.pendingTransaction.delete(tx.transactionHash)
+      })
+  }
+
+  async stop() {
```

Comment:
added stop function for handling all txs properly.
without stop functtion txmgr execute can not properly handle all the txs.